cerbero/build/source.py

```diff
@@ -408,14 +411,93 @@ class Git (GitCache):
     Source handler for git repositories
     '''
 
+    # Maps recipes with common sources share to the same fetch_done condition.
+    _fetch_done_cond_map = {}
+
     def __init__(self):
         GitCache.__init__(self)
         if self.commit is None:
             # Used by recipes in recipes/toolchain/
             self.commit = 'origin/sdk-%s' % self.version
+
+        self._update_fetch_done_cond()
+        self._fetching = False
+
         # For forced commits in the config
         self.commit = self.config.recipe_commit(self.name) or self.commit
 
+    def _get_source_id(self):
+        return (self.repo_dir, tuple(self.remotes.items()))
+
+    def _update_fetch_done_cond(self):
+        source_id = self._get_source_id()
+        if source_id in Git._fetch_done_cond_map:
+            return
+
+        lock = asyncio.Lock()
+        Git._fetch_done_cond_map[source_id] = asyncio.Condition(lock)
+
+    def _get_fetch_done_cond(self):
+        source_id = self._get_source_id()
+        return Git._fetch_done_cond_map[source_id]
+
+    def _get_git_cache_recipes(self):
+        recipes = self.config.cookbook.get_recipes_list()
+        return iter(r for r in recipes if r.stype == SourceType.GIT)
```

Comment:
`GitExtractedTarball` inherits from `Git`. So, we need to use `issubclass(r.stype, SourceType.GIT)` rather than checking only for `SourceType.GIT`. However, I've seen some checks checking explicitly for `SourceType.GIT or SourceType.GIT_TARBALL`, so maybe we should use that instead. IMHO,I prefer consistency and whatever we do, we modify it so that either everything checks for inheritance, or everything checks for one or the other stype.

Reply:
I think there's some magic metaclass, that it makes the recipe to automagically inherit from the corresponding value (a class) set in stype (which may be SourceType.GIT, SourceType.GIT_TARBALL, ...). So I think it's the same, but not sure :-)

Reply:
Ah... this is true, but I should use `issubclass(r.stype, SourceType.GIT)`, yes.